packages/output/src/browser/output-widget.tsx

```diff
@@ -14,137 +14,238 @@
-                }
+    get isLocked(): boolean {
+        return !!this.state.locked;
+    }
+
+    setInput(channelName: string): void {
+        this.outputChannelManager.getChannel(channelName).setVisibility(true);
+    }
+
+    protected revealLastLine(): void {
+        if (this.isLocked) {
+            return;
+        }
+        const editor = this.editor;
+        if (editor) {
+            const model = editor.getControl().getModel();
+            if (model) {
+                const lineNumber = model.getLineCount();
+                const column = model.getLineMaxColumn(lineNumber);
+                editor.getControl().revealPosition({ lineNumber, column }, monaco.editor.ScrollType.Smooth);
             }
         }
-        if (result.length === 0) {
-            result.push(<div className={'theia-output-lines'} key={id++}>{'<no output yet>'}</div>);
+    }
+
+    private get selectedChannel(): OutputChannel | undefined {
+        return this.outputChannelManager.selectedChannel;
+    }
+
+    private async createEditorWidget(): Promise<EditorWidget | undefined> {
+        if (!this.selectedChannel) {
+            return undefined;
         }
-        return result;
+        const { name } = this.selectedChannel;
+        const editor = await this.editorProvider.get(OutputUri.create(name));
+        return new EditorWidget(editor, this.selectionService);
     }
 
-    protected onUpdateRequest(msg: Message): void {
-        super.onUpdateRequest(msg);
-        setTimeout(() => {
-            const div = document.getElementById(OutputWidget.IDs.CONTENTS) as HTMLDivElement;
-            if (div && div.children.length > 0) {
-                div.children[div.children.length - 1].scrollIntoView(false);
+    private get editor(): MonacoEditor | undefined {
+        for (const widget of toArray(this.editorContainer.children())) {
+            if (widget instanceof EditorWidget) {
+                if (widget.editor instanceof MonacoEditor) {
+                    return widget.editor;
+                }
             }
-        });
+        }
+        return undefined;
     }
+
 }
 
 export namespace OutputWidget {
-    export namespace IDs {
-        export const CONTENTS = 'outputContents';
-        export const CHANNEL_LIST = 'outputChannelList';
+    export interface State {
+        locked?: boolean;
     }
 }
+
+/**
+ * @deprecated Use `OutputWidget.ID` instead.
+ */
+export const OUTPUT_WIDGET_KIND = OutputWidget.ID;
+
+/**
+ * Customized `DockPanel` that does not allow dropping widgets into it.
+ * Intercepts `'p-dragover'` events, and sets the desired drop action to `'none'`.
+ */
+class NoopDragOverDockPanel extends DockPanel {
+
+    constructor(options?: DockPanel.IOptions) {
+        super(options);
+        NoopDragOverDockPanel.prototype['_evtDragOver'] = (event: IDragEvent) => {
+            event.preventDefault();
+            event.stopPropagation();
+            event.dropAction = 'none';
+        };
+    }
+
+}
```

Comment:
@kittaakos, it looks like this code is responsible for #10932. Do you recall the motivation for making the Output view undroppable?

Reply:
Thanks for taking care of this!

> it looks like this code is responsible for #10932.

Do you want to have the same behavior as the _Problem_ view? Split the view like this:

https://user-images.githubusercontent.com/1405703/172688966-70e94802-d816-46f0-9dab-9811712cef70.mp4

It is not possible with the _Output_ view; it's a bug.
 
The main reason I used a custom panel implementation was to avoid dropping an editor into the panel. Like this:

https://user-images.githubusercontent.com/1405703/172689342-a5101c63-3a55-4642-a38d-2f9db9bea451.mp4

I hope this helps. Let me know if there is anything I can assist you with fixing this.

Reply:
Thanks for the screen casts; they help clarify things quite a bit. I didn't realize the Output widget behaved as a panel with each of the channels as a child widget. I think it would be nice if the Output view could allow drops as other widgets do, to either place the new widget on the bottom tabbar or split the bottom pane. Otherwise, it feels like a bug, even though it's deliberate, since it isn't clear to users why the Output widget would be different from other widgets in this respect.

Reply:
It looks like something like this might do the trick:

```typescript

/**
 * Customized `DockPanel` that does not allow dropping widgets into it.
 * Intercepts `'p-dragover'` events, and sets the desired drop action to `'none'`.
 */
class NoopDragOverDockPanel extends DockPanel { }
const noop = () => { };
NoopDragOverDockPanel.prototype['_evtDragOver'] = noop;
NoopDragOverDockPanel.prototype['_evtDrop'] = noop;
NoopDragOverDockPanel.prototype['_evtDragLeave'] = noop;
```
By preventing the Output widget from handling any of those events, we pass them up to the bottom pane dock panel and allow it to handle them. What do you think of that?